### ISSUE TITLE
CustomEvent is not a constructor error after upgrading to 0.15

### ISSUE THREAD

**Issue by jesper-johansson** [webpro/DOMtastic#69]
This line gives an error:
`child.dispatchEvent(new CustomEvent('e', {bubbles: true}));` in `trigger.js:169`

Polyfill needed?

**Comment by webpro** [comment]
In what browser or environment are you getting this error?

There is a [polyfill](https://github.com/webpro/DOMtastic/blob/b727c24cc614ef3a3dffcbc26832939e06b56334/src/event/trigger.js#L133-L152), but only for environments having a `window` and `window.document`.

**Comment by alexandernanberg** [comment]
I'm seeing this error as well, in both Chrome 59 and Safari etc. We're using webpack to bundle our code.

```js
// example.js
import $ from 'domtastic';

$('body').on('click', 'h1', (e) => {
  console.log(e);
});

```
throws this error
![image](https://user-images.githubusercontent.com/8997319/27629270-1734b0f4-5bf2-11e7-99ef-4ebca361fb4e.png)

**Comment by webpro** [comment]
Could it be that this is broken since DOMtastic v0.13? (change: https://github.com/webpro/DOMtastic/commit/8710c247bf4a1893d396300c42b624d937a5a16b#diff-34a6d62af0cf0b784f8444529f3130ef).

Not sure what could cause this issue in the changes in the 0.15 release. I've changed the `esnext:main` property to `module` in package.json, which should maybe be `esnext` instead. Not sure how Webpack handles this these days. But even if it picks up the bundled/transpiled version it shouldn't be an issue, I guess.

**Comment by alexandernanberg** [comment]
Dont't think so, works with version ~0.14

It seems to be a problem with the `module` property, like you said, Webpack takes the entry from the `module` (which is not transpiled) property and not the `main`. The event error occurs because  `CustomEvent` is an arrow function so you can't call as a constructor with `new`. Babel transpiles it into a regular function with a `.bind()` and then you can call it as a constructor. 

Been reading a bit about how to handle this problem which seem to be pretty widespread but it doesn't look like anyone have solved it completely yet. 

Similar issues here https://github.com/nfl/react-helmet/issues/190, https://github.com/webpack/webpack/issues/2902

**Comment by webpro** [comment]
Thanks, @alexandernanberg. I'm starting to realize that `module` is only about the *module* syntax (ESM), but the source code of this lib has (much) more ES6 syntax indeed. [@rauschma proposes](http://2ality.com/2017/06/pkg-esnext.html) to use the `esnext` property, but this doesn't seem to be supported by Webpack (yet).

I've just released v0.15.1 with `module` being removed (actually reverted back to `esnext:main`, in case some still rely on it).

**Comment by alexandernanberg** [comment]
@webpro Great, tested again and now it works. Thanks for the quick fix! 

I think this issue can be closed now

**Comment by webpro** [comment]
Thanks!

**Comment by magicspon** [comment]
Hello,

I'm getting this error in 0.15.3
![image](https://user-images.githubusercontent.com/3268717/32982910-b289fc60-cc83-11e7-82cc-274b5b6429b4.png)

any thoughts?

**Comment by webpro** [comment]
I think I've fixed this in [v0.15.4](https://github.com/webpro/DOMtastic/releases/tag/0.15.4).
